src/rdb.c

```diff
@@ -2719,40 +2719,48 @@ void rdbLoadProgressCallback(rio *r, const void *buf, size_t len) {
     }
 }
 
+/* Save the given functions_ctx to the rdb.
+ * The err output parameter is optional and will be set with relevant error
+ * message on failure, it is the caller responsibility to free the error
+ * message on failure. */
 int rdbFunctionLoad(rio *rdb, int ver, functionsCtx* functions_ctx, sds *err) {
     UNUSED(ver);
     sds name = NULL;
     sds engine_name = NULL;
     sds desc = NULL;
     sds blob = NULL;
     uint64_t has_desc;
+    sds error = NULL;
     int res = C_ERR;
     if (!(name = rdbGenericLoadStringObject(rdb, RDB_LOAD_SDS, NULL))) {
-        *err = sdsnew("Failed loading function name");
+        error = sdsnew("Failed loading function name");
         goto error;
     }
 
     if (!(engine_name = rdbGenericLoadStringObject(rdb, RDB_LOAD_SDS, NULL))) {
-        *err = sdsnew("Failed loading engine name");
+        error = sdsnew("Failed loading engine name");
         goto error;
     }
 
     if ((has_desc = rdbLoadLen(rdb, NULL)) == RDB_LENERR) {
-        *err = sdsnew("Failed loading function description indicator");
+        error = sdsnew("Failed loading function description indicator");
         goto error;
     }
 
     if (has_desc && !(desc = rdbGenericLoadStringObject(rdb, RDB_LOAD_SDS, NULL))) {
-        *err = sdsnew("Failed loading function description");
+        error = sdsnew("Failed loading function description");
         goto error;
     }
 
     if (!(blob = rdbGenericLoadStringObject(rdb, RDB_LOAD_SDS, NULL))) {
-        *err = sdsnew("Failed loading function blob");
+        error = sdsnew("Failed loading function blob");
         goto error;
     }
 
     if (functionsCreateWithFunctionCtx(name, engine_name, desc, blob, 0, err, functions_ctx) != C_OK) {
+        if (!error) {
+            error = sdsnew("Failed creating the function");
+        }
```

Comment:
did you mean to test and set `err`?
or pass `error` to `functionsCreateWithFunctionCtx`?
something here seems odd, maybe we don't want to have both variables, or just one internally for the function, and handle the output arg only at the end.

Reply:
This is how it is done, `error` is the local variable that used everywhere and `err` is the output parameter that is set at the end with the value of `error` (if given). Did I missed something?

Reply:
we're good.. it's just that i looked at an intermediate commit.